compiler/src/checkAnnot.ml

```diff
@@ -82,15 +82,20 @@ let check_no_for_loop (_, fds) =
       check_no_for_loop ~funname:f_name.fn_name f_body)
     fds
 
+let is_inlinable ir =
+  match ir with
+  | Ccall _ | Cif _ -> true
+  | _ -> false
+
```

Comment:
I don’t understand this part: if an assignment or an `Copn`  is annotated “inline”, there will be no error. Why?

Reply:
The idea is to move all inline releated stuff to annotations on call sites. The problem is that several passes propagate the `instr_info` of some instructions (Inlining uses the call site's `instr_info` for the assignments before and after the body, Array Initialization uses it to add `ArrayInit` assignments).
One solution would be to add a `consume_inline` field to `InstrInfo` and use it in all these pases. Another one maybe is that when we insert code we use an `instr_info` derived from the location of the one from the call site, but without the annotations. But Benjamin said that the inline annotation makes no sense for instructions other than `Ccall` or `Cif`, so changing checkAnnot means less changes.
If you think this check is important, I'll implement one of the alternatives. Or something else if there are better ways of doing this.

Reply:
Oh, I see. Consuming the annotation sounds better, but might deserve an other PR (up to you).

Reply:
To me it seems reasonable, but consuming exactly the inline annotation in Array Initialization sounds too ad hoc. I asked about this on Zulip.

Reply:
I created #592 and rebased on top of that.